ibis/backends/conftest.py

```diff
@@ -259,9 +441,29 @@ def pytest_runtest_call(item):
 
 
 @pytest.fixture(params=_get_backends_to_test(), scope='session')
-def backend(request, data_directory):
-    """Return an instance of BackendTest."""
+def backend(
+    request, data_directory, script_directory, tmp_path_factory, worker_id
+):
+    """Return an instance of BackendTest, loaded with data."""
+    from filelock import FileLock
+
     cls = _get_backend_conf(request.param)
+
+    # Load testdata once per session
+    if worker_id == "master":  # single-threaded pytest
+        cls.load_data(data_directory, script_directory)
+        return cls(data_directory)
+
+    # multi-threaded pytest
+
+    # get the temp directory shared by all workers
+    root_tmp_dir = tmp_path_factory.getbasetemp().parent
+
+    fn = root_tmp_dir / "lockfile"
+    with FileLock(str(fn) + ".lock"):
+        if not fn.is_file():
+            cls.load_data(data_directory, script_directory)
```

Comment:
Is the lockfile unique across pytest invocations meaning that we load data each time we execute pytest?

Reply:
Oh, good point! The lockfile is unique, and so as long as it exists, we will not load data each time we execute.

Would the preferred behaviour be to:
* keep that behaviour
* clean up the lockfile in teardown